src/main/scala/mesosphere/marathon/core/task/termination/impl/KillServiceDelegate.scala

```diff
@@ -38,4 +40,17 @@ private[termination] class KillServiceDelegate(actorRef: ActorRef) extends KillS
     logger.info(s"Kill and forget following instances for reason $reason: ${instances.map(_.instanceId).mkString(",")}")
     actorRef ! KillInstancesAndForget(instances)
   }
+
+  /**
+    * Begins watching immediately for terminated instances. Future is completed when all instances are seen.
+    */
+  def watchForKilledInstances(instances: Seq[Instance]): Future[Done] = {
```

Comment:
Any reason we are not using `KillServiceActor` instead? We do receive all instance changes there. That seems simpler to me.

Reply:
This method was just moved from _KillServiceActor.scala_  `KillServiceActor` is about retrying resending kills and not sending too many kills at once etc.

Reply:
yeah, it's just so that this method is on the same interface you use for killing. I think it's fine